### PR TITLE
Error messages up4728

### PR DESCRIPTION
**Pull request by chrieke**
Removes explicit error handling and passes through error message. 

HTTPError or ValueError as the default error?

image

### CODE REVIEW

**Comment by rodrigoalmeida94** [comment]
> Removes explicit error handling and passes through error message.
> 
> HTTPError or ValueError as the default error?
> 
> image

I think in the Ideal world we would define our own UP42SDKError class that would be specific for this context. For now we could just raise RequestsException. Using ValueError can cause some confusion if you want to do try: except statements.

File: up42/auth.py [hunk 1]
```diff
@@ -111,10 +111,10 @@ def _endpoint(self) -> str:
     def _get_token(self):
         try:
             self._get_token_project()
-        except requests.exceptions.HTTPError:
+        except requests.exceptions.HTTPError as err:
             raise ValueError(
```
Comment (rodrigoalmeida94) [inline up42/auth.py:115]: What does this `from err` do? This is a new notation for me :)

File: up42/auth.py [hunk 2]
```diff
@@ -224,18 +224,13 @@ def _request(
...
-                )
-            elif response.status_code == 404:
-                raise ValueError("Product not found!")
+        try:
+            response.raise_for_status()
+        except requests.exceptions.RequestException as err:  # Base error class
+            err_message = json.loads(response.text)["error"]
+            err_message = f"{err_message['code']} Error - {err_message['message']}!"
+            logger.error(err_message)
+            raise requests.exceptions.HTTPError(err_message) from err
```
Comment (rodrigoalmeida94) [inline up42/auth.py:233]: Nice!

**Review by rodrigoalmeida94 — APPROVED** [review]
LGTM

**Review by marooned20 — APPROVED** [review]
LGTM!

File: up42/auth.py (hunk 1 above)
Comment (chrieke) [reply up42/auth.py:115]: Similar to the stacktrace in the screenshot example, is a bit more explicit that this custom Exception is a direct consequence of the previous exception. Is a detail but fits well here I think.

**Pull request merged**